tests/tests_unit/test_api/test_data_modeling/test_instances.py

```diff
@@ -33,3 +39,116 @@ def test_instances_api_dump_instance_source(sources, expected):
     # ViewIdentifier = Union[ViewId, Tuple[str, str], Tuple[str, str, str]]
     # ViewIdentifier | Sequence[ViewIdentifier] | View | Sequence[View]
     assert expected == InstancesAPI._dump_instance_source(sources)
+
+
+@pytest.fixture
+def empty_node_dump():
+    return {
+        "space": "space",
+        "externalId": "xid",
+        "version": 1,
+        "lastUpdatedTime": 2,
+        "createdTime": 3,
+    }
+
+
+@pytest.fixture
+def empty_edge_dump():
+    return {
+        "space": "space",
+        "externalId": "xid",
+        "version": 4,
+        "lastUpdatedTime": 5,
+        "createdTime": 6,
+        "type": {"space": "foo-space", "externalId": "bar-xid"},
+        "startNode": {"space": "the-space", "externalId": "the-xid"},
+        "endNode": {"space": "the-space", "externalId": "the-xid"},
+    }
+
+
+@pytest.mark.parametrize("instance_type", (Node, Edge))
+def test_instances__quick_property_access_single_source(
+    instance_type: type[Instance],
+    empty_node_dump: dict[str, Any],
+    empty_edge_dump: dict[str, Any],
+):
+    # Note: 'property' in this test refers to an instance property, not a Python property
+    resource = {Node: empty_node_dump, Edge: empty_edge_dump}[instance_type]
+    resource["properties"] = {"space": {"view/v8": {"prop1": 1, "prop2": "two", "3prop": [1, 2, 3]}}}
+    inst = instance_type.load(resource)
+
+    # Non-property should fail __getitem__ and __setitem__:
+    with pytest.raises(KeyError):
+        inst["space"]
+    with pytest.raises(RuntimeError):
+        inst["space"] = "more-space"
+
+    assert hasattr(inst, "id") is False
+    assert hasattr(inst, "space") is True
+    assert inst.space == "space"
+    inst.space = "more-space"
+    assert inst.space == "more-space"
+
+    # Property should work fine with both access/set/delete protocols:
+    assert inst.prop1 == 1
+    assert inst["prop1"] == 1
```

Comment:
This is more of a personal preference, but I am typically not a big fan of overwriting `__set_attr__`. I know this is done for example in pandas, but it tend to confuse what is an attribute with what is actually data.

Reply:
Agreed, reverted to just "dict" syntax that exclusively targets properties 👍